src/components/Explore/Explore.tsx

```diff
@@ -160,12 +160,30 @@ const ExploreCopy: React.FunctionComponent<{
         ORIGINAL_EXPLORE_METRICS.includes(currentMetric),
     );
 
+    const [period, setPeriod] = useState<Period>(Period.ALL);
+    const allPeriodLabels = getAllPeriodLabels();
+    const dateRange = getDateRange(period);
+
+    const onSelectPeriod = (newPeriod: Period) => {
+      const newPeriodLabel = periodMap[newPeriod].label;
+      setPeriod(newPeriod);
+      trackExploreEvent(EventAction.SELECT, `Period: ${newPeriodLabel}`);
+    };
+
     useEffect(() => {
       setNormalizeData(
         selectedLocations.length > 1 &&
           ORIGINAL_EXPLORE_METRICS.includes(currentMetric),
       );
-    }, [currentMetric, selectedLocations]);
+      setMetricMenuLabel(metricLabels[currentMetric]);
+      setTimeRangeMenuLabel(allPeriodLabels[period]);
+    }, [
+      currentMetric,
+      selectedLocations,
+      metricLabels,
+      allPeriodLabels,
+      period,
+    ]);
```

Comment:
How about defining variables and setting them as dependencies instead?

```tsx
const timeRangeLabel = allPeriodLabels[period];
const currentMenuLabel = metricLabels[currentMetric];
const isNormalizedData = selectedLocations.length > 1 && /* ... */;

useEffect(() => {
  /* update the state using the values */
}, [timeRangeLabel, isNormalized, currentMenuLabel]);
```

I'm not sure if this is actually better, but I feel that having fewer dependencies might help to update the state only when needed